ts/screens/messages/paginated/MessageRouterScreen.tsx

```diff
@@ -0,0 +1,165 @@
+ * @param message
+ * @param messageDetails
+ */
+const navigateToScreenHandler =
+  (message: UIMessage, messageDetails: UIMessageDetails) =>
+  (dispatch: OwnProps["navigation"]["dispatch"]) => {
+    if (euCovidCertificateEnabled && messageDetails.euCovidCertificate) {
+      navigateBack();
+      navigateToEuCovidCertificateDetailScreen({
+        authCode: messageDetails.euCovidCertificate
+          .authCode as EUCovidCertificateAuthCode,
+        messageId: message.id
+      });
+    } else {
+      navigateBack();
+      dispatch(
+        navigateToPaginatedMessageDetailScreenAction({
+          message,
+          messageDetails
+        })
+      );
+    }
+  };
+
+const MessageRouterScreen = ({
+  cancel,
+  cursors,
+  loadMessageDetails,
+  loadPreviousPage,
+  reloadPage,
+  maybeMessage,
+  maybeMessageDetails,
+  messageId,
+  navigation
+}: Props): React.ReactElement => {
+  // used to automatically dispatch loadMessages if the pot is not some at the first rendering
+  // (avoid displaying error at the first frame)
+  const firstRendering = useRef(true);
+  const isLoading = !pot.isError(maybeMessageDetails);
+  const errorMessage = pot.isError(maybeMessageDetails)
+    ? maybeMessageDetails.error
+    : undefined;
+
+  const tryLoadMessageDetails = useCallback(() => {
+    if (maybeMessage === undefined) {
+      if (pot.isNone(cursors)) {
+        // nothing in the collection, refresh
+        reloadPage();
+      } else if (pot.isSome(cursors)) {
+        // something in the collection, get the new ones only
+        loadPreviousPage(cursors.value.previous);
+      }
+    }
+    loadMessageDetails(messageId);
+  }, [
+    maybeMessage,
+    cursors,
+    messageId,
+    loadMessageDetails,
+    loadPreviousPage,
+    reloadPage
+  ]);
+
+  useEffect(() => {
+    // message in the list and its details loaded: green light
+    if (isStrictSome(maybeMessageDetails) && maybeMessage !== undefined) {
+      navigateToScreenHandler(
+        maybeMessage,
+        maybeMessageDetails.value
+      )(navigation.dispatch);
+      return;
+    }
+    if (firstRendering.current) {
+      tryLoadMessageDetails();
+      // eslint-disable-next-line functional/immutable-data
+      firstRendering.current = false;
+    }
+  }, [
+    loadMessageDetails,
+    maybeMessage,
+    maybeMessageDetails,
+    messageId,
+    navigation,
+    tryLoadMessageDetails
+  ]);
+
+  return (
+    <LoadingErrorComponent
+      errorText={errorMessage}
+      isLoading={isLoading}
+      loadingCaption={I18n.t("messageDetails.loadingText")}
+      onAbort={cancel}
+      onRetry={tryLoadMessageDetails}
+    />
+  );
+};
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  cancel: () => navigateBack(),
+  loadMessageDetails: (id: UIMessageId) => {
+    dispatch(loadMessageDetails.request({ id }));
+  },
+  loadPreviousPage: (cursor?: Cursor) =>
+    dispatch(loadPreviousPageMessages.request({ cursor, pageSize: 100 })),
```

Comment:
```suggestion
    dispatch(loadPreviousPageMessages.request({ cursor, pageSize: 100 })),
```
my obsession 😆 , avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming))
we should use the config const instead

Reply:
Just realized I made a mistake committing the other suggestion.
This is not the _pageSize_ from config: that thing is used for actual pagination.
This is instead the maximum amount supported by the APIs. Note that this is pretty standard and it will default to this number if not specified.

All I can do is to move also this number to `ts/config.ts`.

Reply:
Updated [here](https://github.com/pagopa/io-app/pull/3580/commits/5483199366753b094a6034819992105e6a4769ad) using a constant for the _magic number_ 😄